pkg/db/db.go

```diff
@@ -83,7 +85,50 @@ func (o Organization) KeysAsSet() (*jwk.Set, error) {
 	for _, key := range o.Keys {
 		maps = append(maps, key.(map[string]interface{}))
 	}
-	return crypto.MapsToJwkSet(maps)
+	result, err := crypto.MapsToJwkSet(maps)
+	if err != nil {
+		return nil, err
+	}
+	// Support deprecated PublicKey
+	if o.PublicKey != nil {
+		key, err := crypto.PemToPublicKey([]byte(*o.PublicKey))
+		if err != nil {
+			return nil, err
+		}
+		pubKey, err := jwk.New(key)
+		if err != nil {
+			return nil, err
+		}
+		result.Keys = append(result.Keys, pubKey)
+	}
+	return result, nil
+}
+
+// HasKey checks whether the given key is owned by the organization
+func (o Organization) HasKey(key jwk.Key, instant time.Time) (bool, error) {
+	// func can't return error
+	keyTp, _ := key.Thumbprint(crypto2.SHA256)
+	keys, err := o.KeysAsSet()
+	if err != nil {
+		return false, err
+	}
+	for _, k := range keys.Keys {
+		// func can't return error
+		tp, _ := k.Thumbprint(crypto2.SHA256)
+		if bytes.Compare(keyTp, tp) == 0 {
+			// Found the key
+			chainInterf, chainExists := k.Get("x5c")
+			if chainExists {
+				certificate := chainInterf.([]*x509.Certificate)[0]
```

Comment:
are there always 1 item in this array?

Reply:
When the member "x5c" exists, the JWK library guarantees there's at least 1 cert. entry. The topmost is our most significant certificate which we'll be checking.